src/service/topics.js

```diff
@@ -80,11 +80,33 @@ const getSubscribeableTopic = async () => {
   }
 };
 
+const getWhoToFollowList = async (topics, locations, page = 1, axiosOptions = {}) => {
+  try {
+    const topicsQueryString = topics
+      .map((topic) => `topics[]=${encodeURIComponent(topic)}`)
+      .join('&');
+    const locationsQueryString = locations
+      .map((topic) => `locations[]=${encodeURIComponent(topic)}`)
+      .join('&');
+    console.warn(
+      `/who-to-follow/list_v2?${locationsQueryString}&${topicsQueryString}&page=${page}`
+    );
+    const result = await api.get(
+      `/who-to-follow/list_v2?${locationsQueryString}&${topicsQueryString}&page=${page}`,
+      axiosOptions
+    );
+    return result.data.body;
+  } catch (e) {
+    throw new Error(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getWhoToFollowList` is missing error handling for the case when the API call fails. It's good practice to handle such scenarios and provide a meaningful error message or fallback mechanism. Also, it seems like you are using `console.warn` for debugging purposes. It's not recommended to leave debug logs in production code.

```diff
const getWhoToFollowList = async (topics, locations, page = 1, axiosOptions = {}) => {
  try {
    const topicsQueryString = topics
      .map((topic) => `topics[]=${encodeURIComponent(topic)}`)
      .join('&');
    const locationsQueryString = locations
      .map((topic) => `locations[]=${encodeURIComponent(topic)}`)
      .join('&');
-91:     console.warn(
-92:       `/who-to-follow/list_v2?${locationsQueryString}&${topicsQueryString}&page=${page}`
-93:     );
    const result = await api.get(
      `/who-to-follow/list_v2?${locationsQueryString}&${topicsQueryString}&page=${page}`,
      axiosOptions
    );
    return result.data.body;
  } catch (e) {
-100:     throw new Error(e);
+100:     console.error('Failed to fetch who to follow list:', e);
+101:     return [];
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->